main.tf

```diff
@@ -46,7 +46,7 @@ resource "argocd_application" "this" {
     namespace = var.argocd_namespace
   }
 
-  wait = true
+  wait = var.app_autosync == { "allow_empty" = tobool(null), "prune" = tobool(null), "self_heal" = tobool(null) } ? false : true
```

Comment:
I still don't see the need to use the `tobool(null)` since it will automatically return `null`. Can someone enlighten me?